pkg/reconciler/route/route.go

```diff
@@ -269,19 +270,38 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 		return err
 	}
 
-	logger.Info("Creating ClusterIngress.")
-	desired, err := resources.MakeClusterIngress(ctx, r, traffic, tls, resources.GetNames(clusterLocalServices), ingressClassForRoute(ctx, r))
+	// reconcile ingress and it's children resources
+	_, err = c.reconcileIngressResources(ctx, r, traffic, tls, clusterLocalServiceNames, ingressClassForRoute(ctx, r),
+		&ClusterIngressResources{
+			BaseIngressResources: BaseIngressResources{
+				servingClientSet: c.ServingClientSet,
+			},
+			clusterIngressLister: c.clusterIngressLister,
+		},
+	)
+
 	if err != nil {
 		return err
 	}
-	clusterIngress, err := c.reconcileClusterIngress(ctx, r, desired)
+
+	// reconcile ingress and it's children resources
+	ingress, err := c.reconcileIngressResources(ctx, r, traffic, tls, clusterLocalServiceNames, ingressClassForRoute(ctx, r),
+		&IngressResources{
+			BaseIngressResources: BaseIngressResources{
+				servingClientSet: c.ServingClientSet,
+			},
+			ingressLister: c.ingressLister,
+		},
+	)
+
 	if err != nil {
 		return err
 	}
-	r.Status.PropagateClusterIngressStatus(clusterIngress.Status)
+
+	r.Status.PropagateIngressStatus(*ingress.GetStatus())
```

Comment:
We have two status (from ClusterIngress and Ingress).   We propagate the status from Ingress.

Reply:
Can we turn this into a code comment?